### PR TITLE
feat(es): respect format parameter in range query

### PR DESCRIPTION
**Pull request by etolbakov**
### Description

Introduce the support for `format` parameter in `RangeQuery`

### How was this PR tested?

```sh
curl -X GET "http://localhost:7280/api/v1/_elastic/stackoverflow-schemaless/_search?pretty" -H 'Content-Type: application/json' -d'
{
  "query": {
    "range": {
      "age": {
        "gte": "2023-01-01",
        "lte": "2023-12-31",
        "boost": 2.0,
        "format": "%Y-%m-%d"
      }
    }
  }
}
```

### Related issue
https://github.com/quickwit-oss/quickwit/issues/5109

### CODE REVIEW

File: quickwit/quickwit-query/src/elastic_query_dsl/range_query.rs [hunk 1]
```diff
@@ -81,3 +94,15 @@ impl ConvertableToQueryAst for RangeQuery {
         Ok(ast.boost(boost))
     }
 }
+
+
+fn parse_and_convert(value: JsonLiteral, fmt: &str) -> anyhow::Result<JsonLiteral> {
```
Comment (fulmicoton) [inline quickwit/quickwit-query/src/elastic_query_dsl/range_query.rs:99]: I think there is benefit to parsing the time input format in caller.
```suggestion
fn parse_and_convert(value: JsonLiteral, date_format: &DateTimeInputFormat) -> anyhow::Result<JsonLiteral> {
```

**Comment by etolbakov** [comment]
Thanks for the reply @fulmicoton 
At the moment my understanding is lacking 2 things
1) General logic:  if `format` is specified - we use it to convert `gt`/ `gte`, `lt`/`lte` to date  and then the result convert to a `JsonLiteral::String`. Please let me know if I'm wrong.

2) `DateTimeInputFormat` conversion: there's a [from_str](https://github.com/quickwit-oss/quickwit/blob/4de8d7afb4bebf21ba95a93101856db388128b0b/quickwit/quickwit-datetime/src/date_time_format.rs#L188) method that I was planing to use, but passing `"format": "yyyy-MM-dd"` won't succeed. Am I doing wrong assumptions on input format?

**Comment by fulmicoton** [comment]
1.  General logic: I think your approach is ok.

File: quickwit/quickwit-query/src/elastic_query_dsl/range_query.rs (hunk 1 above)
Comment (etolbakov) [reply quickwit/quickwit-query/src/elastic_query_dsl/range_query.rs]: Great point! it's a shame that I didn't notice it myself. Thanks!

**Comment by etolbakov** [comment]
@fulmicoton thank you for the response!
Could you please take a look a the changes when you have time?
If that LGTY , I will add a test (not sure where though) and convert it to the PR.

<details>
  <summary>Examples Successful and Failed Responses ⬇️ </summary>
  
  ```sh
curl -X GET "http://localhost:7280/api/v1/_elastic/stackoverflow-schemaless/_search?pretty" -H 'Content-Type: application/json' -d'
{
  "query": {
    "range": {
      "age": {
        "gte": "2023-01-01",
        "lte": "2023-12-31",
        "boost": 2.0,
        "format": "yyyy-MM-dd"
      }
    }
  }
}
'
{
  "status": 400,
  "error": {
    "caused_by": null,
    "reason": "Failed to parse date time: a character literal was not valid",
    "stack_trace": null,
    "type": null
  }
}


curl -X GET "http://localhost:7280/api/v1/_elastic/stackoverflow-schemaless/_search?pretty"
-H 'Content-Type: application/json' -d'
∙ {
∙   "query": {
∙     "range": {
∙       "age": {
∙         "gte": "2023-01-01",
∙         "lte": "2023-12-31",
∙         "boost": 2.0,
∙         "format": "%Y-%m-%d"
∙       }
∙     }
∙   }
∙ }
∙ '
{
  "took": 5,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 0,
      "relation": "eq"
    },
    "hits": []
  }
}
  ```
</details>

**Review by fulmicoton — APPROVED** [review]

**Comment by fulmicoton** [comment]
thank you! I will complete our rest api test suite in another PR and send it to you for next time!

**Pull request merged**